### PR TITLE
Use latest version of play services.

### PR DESCRIPTION
**Pull request by FME-nstokoe**
In order to be compatible with usage of the google analytics cordova plugin:

https://github.com/danwilson/google-analytics-plugin

### CODE REVIEW

File: src/android/build-extras.gradle [hunk 1]
```diff
@@ -13,7 +13,7 @@ dependencies {
     compile 'com.android.support:support-v4:23.4.0'
     compile 'com.android.support:cardview-v7:23.4.0'
     compile 'com.android.support:support-annotations:23.4.0'
-    compile 'com.google.android.gms:play-services-location:9.0.2'
+    compile 'com.google.android.gms:play-services-location:+'
```
Comment (macdonst) [inline src/android/build-extras.gradle]: I would make this line `9.0.2+` to ensure a minimum version of GPS.

**Comment by rlepinski** [comment]
What error are you receiving when trying to use both builds? I am curious how dependencies are managed when they are defined in the plugin.xml instead of a build.gradle file. Usually gradle will resolve dependency conflicts by picking the newest version. If the plulgin.xml does not use gradle, then I would assume its pulling in the artifact directly and we might run into duplicate symbol issues.

**Comment by macdonst** [comment]
@rlepinski I took a quick look and it uses:

```
<framework src="com.google.android.gms:play-services-analytics:+" />
```

the two framework tags should end up writing build.gradle properly but I haven't tested it yet.

**Comment by rlepinski** [comment]
So google analytics is newer then the rest of Google Play Service? The above patch will only change location, but what you don't see is it also uses GCM at 9.0.2 - https://github.com/urbanairship/android-library/blob/master/urbanairship-sdk/build.gradle#L46. Are we sure this is actually causing issues?

**Comment by FME-nstokoe** [comment]
The error we receive when we try to compile our android app with Ionic is:
```
UNEXPECTED TOP-LEVEL EXCEPTION:
com.android.dex.DexException: Multiple dex files define Lcom/google/android/gms/location/places/PlaceReport;
	at com.android.dx.merge.DexMerger.readSortableTypes(DexMerger.java:579)
	at com.android.dx.merge.DexMerger.getSortedTypes(DexMerger.java:535)
	at com.android.dx.merge.DexMerger.mergeClassDefs(DexMerger.java:517)
	at com.android.dx.merge.DexMerger.mergeDexes(DexMerger.java:164)
	at com.android.dx.merge.DexMerger.merge(DexMerger.java:188)
	at com.android.dx.command.dexer.Main.mergeLibraryDexBuffers(Main.java:504)
	at com.android.dx.command.dexer.Main.runMonoDex(Main.java:334)
	at com.android.dx.command.dexer.Main.run(Main.java:277)
	at com.android.dx.command.dexer.Main.main(Main.java:245)
	at com.android.dx.command.Main.main(Main.java:106)


 FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':transformClassesWithDexForRelease'.
> com.android.build.api.transform.TransformException: com.android.ide.common.process.ProcessException: org.gradle.process.internal.ExecException: Process 'command '/Library/Java/JavaVirtualMachines/1.6.0.jdk/Contents/Home/bin/java'' finished with non-zero exit value 2

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.
```

File: src/android/build-extras.gradle (hunk 1 above)
Comment (FME-nstokoe) [inline src/android/build-extras.gradle]: Using ```9.0.2+``` also breaks the build. See the error below.

**Comment by rlepinski** [comment]
Could you provide any *.gradle files you see in the generated android application?

**Comment by FME-nstokoe** [comment]
From the failed build or the successful build using the fix?

**Comment by rlepinski** [comment]
doing just `+` fixes the build? If you have both available that would be great, but if not the broken build gradle files.

**Comment by FME-nstokoe** [comment]
[gradle_files_from_failed_build.zip](https://github.com/urbanairship/phonegap-ua-push/files/366391/gradle_files_from_failed_build.zip)
[gradle_files_from_successful_build.zip](https://github.com/urbanairship/phonegap-ua-push/files/366392/gradle_files_from_successful_build.zip)

**Comment by rlepinski** [comment]
Thanks!

I am a little nervous to do just `+`, can we compromise and do `9.+`? They sometimes add breaking changes to major versions.

**Comment by rlepinski** [comment]
Also can I get you to fill out http://docs.urbanairship.com/contribution-agreement.html? We need it even for small changes like these.

**Comment by macdonst** [comment]
@rlepinski if you don't use at least 9.0.2 you run into the possibility of the dreaded:

```
Fatal Exception: java.lang.IncompatibleClassChangeError: android.support.v4.content.ContextCompat
       at com.google.android.gms.iid.zzd.zzeC(Unknown Source)
       at com.google.android.gms.iid.zzd.<init>(Unknown Source)
       at com.google.android.gms.iid.zzd.<init>(Unknown Source)
       at com.google.android.gms.iid.InstanceID.zza(Unknown Source)
       at com.google.android.gms.iid.InstanceID.getInstance(Unknown Source)
       at com.adobe.phonegap.push.PushPlugin$1.run(PushPlugin.java:75)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
       at java.lang.Thread.run(Thread.java:841)
```

As Google screwed up releasing 9.0.0 and I don't think the fix was in until 9.0.2

**Comment by rlepinski** [comment]
@macdonst The UA SDK requires a min version of 9.0.2 for GCM, so we can expect the min version  available to be 9.0.2. I am testing `[9.0.2,10[` as the version -http://ant.apache.org/ivy/history/2.1.0/settings/version-matchers.html. Ill report back once I update my Android SDK.

**Comment by macdonst** [comment]
@rlepinski cool, just wanna make sure you don't run into the same pain that I did.

**Comment by rlepinski** [comment]
@macdonst We have  :( - http://stackoverflow.com/questions/37322365/android-urbanairship-crash-in-takeoff

@FME-nstokoe 

Could you update the PR to do: 
```
compile 'com.google.android.gms:play-services-gcm:[9.0.2,10['
compile 'com.google.android.gms:play-services-location:[9.0.2,10['
```

Then both gcm and location will be at the same version.

**Comment by FME-nstokoe** [comment]
Ok, let me try this new setting and see if it works. Assuming it does I will update the pull request.

I've signed the contributor agreement.

**Comment by FME-nstokoe** [comment]
@rlepinski 
That fix looks good. Thanks for your help.

**Comment by rlepinski** [comment]
@FME-nstokoe Ill do a full release on Monday. Thanks for the PR!

**Pull request merged**